### ISSUE TITLE
New feature: document target parameters

### ISSUE THREAD

**Issue by cvubrugier** [agrover/targetcli-fb#48]
The `get attribute` and `get parameter` commands in targetcli do not provide much information because the description is always the same: _The foobar attribute/parameter_. I would like to improve this situation and I have started some work in this direction. For instance, targetcli `get parameter` in the TPG context would display:

    InitialR2T=Yes
    --------------
    If set to No, the default use of R2T (Ready To Transfer) is disabled.
    
    MaxBurstLength=262144
    ---------------------
    Maximum SCSI data payload in bytes in a Data-In or a solicited Data-Out iSCSI sequence.

I have created a branch named [document-parameters](https://github.com/cvubrugier/targetcli-fb/tree/document-parameters) that shows a proof of concept implementation. Basically, subclasses of `UIRTSLibNode` can declare `attributes` and `parameters` class members to document their configfs parameters. As an illustration, I have added description for the TPG parameters. My implementation is a bit hackish though.

@agrover and others, what do you think about this feature? Is it worthwhile and how can I improve it?

P.S. : my targetcli "document-parameters" branch depends on some changes published in my configshell-fb "document-parameters" branch.

**Comment by agrover** [comment]
Thanks for tackling this! I think the overall approach is the right one.

I think it may be possible to eliminate ui_get_config_group_param_info. Would something like 
```
type, desc = getattr(self, "ui_desc_parameters", {}).get(parameter, ("string", ""))
```
 accomplish the same thing? (basically just taking advantage of getattr and dict.get() letting us provide defaults if the thing is not present.)

thanks again, looking fwd! :smiley: :+1:

**Comment by cvubrugier** [comment]
>      type, desc = getattr(self, "ui_desc_parameters", {}).get(parameter, ("string", ""))

That's much better, thank you!

**Comment by cvubrugier** [comment]
Feature merged in 5025acdba6b0187231555e0391dbce91a4d2a263
